### DISCUSSION TITLE
Manipulation of inputParameters before their input

### DISCUSSION THREAD

**Discussion by JCHacking** [Netflix/conductor#3610] [Q&A] [answered]
Hi, Maybe this is a very novice doubt, but I am new in Conductor and I have the following need.

I have a task which returns an output as follows:
```json
{
    "person1": {
        "name": "NAME",
        "lastname": ["lastname1", "lastname2"]
    },
    "person2": {
        "name": "NAME2"
    }
}
```

How could I get the inputParameters in another task to be like this:
```json
{
    "names": ["NAME", "lastname1", "lastname2", "NAME2"]
}
```

It may also be that person2.name2 does not exist, so that concatenation would be optional.

I have tried SpEL actions like the following: `"[${task1.output.person1.name}] + ${task1.output.person1.lastname} + [task1.output.person2!.name!]"` but I do not get the desired output.

**Comment by c4lm** [comment]
Try this:
```
{
    "names": ["${task1.output.person1.name}", "${task1.output.person1.lastname[0]}", "${task1.output.person1.lastname[1]}", "${task1.output.person2.name}"]
}
```

**Reply by JCHacking** [reply]
The problem is that lastname would be a dynamic list, I would not know exactly the number of elements.
So this solution does not work for me

**Reply by c4lm** [reply, marked as answer]
Then you'd need to have an INLINE task to transform, since your case is not supported by ParameterUtils
